datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVHelper.scala

```diff
@@ -320,13 +322,16 @@ object MVHelper {
   }
 
   def getUpdatedName(name: String): String = {
-    val updatedName = name.replace("(", "_")
+    var updatedName = name.replace("(", "_")
       .replace(")", "")
       .replace(" ", "_")
       .replace("=", "")
       .replace(",", "")
       .replace(".", "_")
       .replace("`", "")
+    if (updatedName.length >= CarbonCommonConstants.MAXIMUM_CHAR_LENGTH) {
+      updatedName = updatedName.substring(0, 110) + CarbonCommonConstants.UNDERSCORE + sysNanoTime
```

Comment:
i think, if you take the substring and add always the same time to it, there can be chance that two columns get same name

Reply:
Take one counter and increment it.

Reply:
done